taskcat/_common_utils.py

```diff
@@ -27,20 +31,19 @@ def name_from_stack_id(stack_id):
     return stack_id.split(":")[5].split("/")[1]
 
 
-def s3_url_maker(bucket, key, client_factory):
-    s3_client = client_factory.get("s3")
+def s3_url_maker(bucket, key, s3_client):
     location = s3_client.get_bucket_location(Bucket=bucket)["LocationConstraint"]
     url = (
         f"https://{bucket}.s3.amazonaws.com/{key}"
     )  # default case for us-east-1 which returns no location
     if location:
-        domain = get_s3_domain(location, client_factory)
+        domain = get_s3_domain(location)
         url = f"https://{bucket}.s3-{location}.{domain}/{key}"
     return url
 
 
-def get_s3_domain(region, client_factory):
-    ssm_client = client_factory.get("ssm")
+def get_s3_domain(region, ssm_client=None):
+    ssm_client = ssm_client if ssm_client else boto3.client("ssm")
```

Comment:
Standalone `boto3.client` versus the `region.client("ssm")`  👎

Reply:
as this is querying public parameters, it does not require any specific account/region

Reply:
Yes, but the default credential set could simply have enough permissions to assume a role.

Reply:
agree that the default set should be fine, but then I need to pass in a client factory instance, which (unless I'm missing something) I may not have at this stage of execution.